### ISSUE TITLE
`truthy` rule false positive

### ISSUE THREAD

**Issue by Pandapip1** [adrienverge/yamllint#671]
Having a key named `on` causes the truthy rule to fail.

Example:

```
name: Continuous Integration

on:  # Error occurs here
  push:
    branches:
      - main
```

**Comment by Pandapip1** [comment]
Any reason why this was immediately closed?

**Comment by andrewimeson** [comment]
This issue has been duplicately reported a bunch of times.

**Comment by Pandapip1** [comment]
> This issue has been duplicately reported a bunch of times.

FWW I did a search for open issues and couldn't find it. Sorry for the inconvenience!
